src/main/java/org/c4marathon/assignment/product/domain/repository/ProductQueryRepository.java

```diff
@@ -0,0 +1,44 @@
+package org.c4marathon.assignment.product.domain.repository;
+
+import static org.c4marathon.assignment.global.utils.PageUtil.SMALL_PAGE_SIZE;
+import static org.c4marathon.assignment.product.domain.QProduct.product;
+import static org.c4marathon.assignment.transaction.domain.QTransaction.transaction;
+
+import java.util.List;
+
+import org.c4marathon.assignment.product.dto.ProductDto;
+import org.springframework.stereotype.Repository;
+
+import com.querydsl.core.BooleanBuilder;
+import com.querydsl.core.types.Projections;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+
+import lombok.RequiredArgsConstructor;
+
+@Repository
+@RequiredArgsConstructor
+public class ProductQueryRepository {
+
+    private final JPAQueryFactory queryFactory;
+
+    public List<ProductDto> getProducts(Long productCursorId, String searchKeyword) {
+        BooleanBuilder dynamicLtProductId = new BooleanBuilder();
+
+        if (productCursorId != null) {
+            dynamicLtProductId.and(transaction.id.lt(productCursorId));
+        }
+
+        return queryFactory
+                .select(Projections.constructor(ProductDto.class,
+                        product.id,
+                        product.productName,
+                        product.price,
+                        product.description))
+                .from(product)
+                .where(dynamicLtProductId
+                        .and(product.productName.like("searchKeyword%")))
```

Comment:
```suggestion
                        .and(product.productName.like(searchKeyword+"%")))
```
searchKeyword는 String 바깥으로 빼야되겠습니다.